parlai/agents/bert_ranker/helpers.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+from parlai.core.torch_ranker_agent import TorchRankerAgent
+import torch
+try:
+    from pytorch_pretrained_bert.modeling import BertLayer, BertConfig
+    from pytorch_pretrained_bert import BertAdam
+    from pytorch_pretrained_bert import BertModel  # NOQA
+except ImportError:
+    raise ImportError(("BERT rankers needs pytorch-pretrained-BERT installed. \n "
+                       "pip install pytorch-pretrained-bert"))
+from parlai.core.utils import _ellipse
+
+
+def add_common_args(parser):
+    """Add command line arguments for this agent."""
+    TorchRankerAgent.add_cmdline_args(parser)
+    parser = parser.add_argument_group('Bert Ranker Arguments')
+    parser.add_argument('--num-samples', type=int, default=131800,
+                        help='Number of samples in the task (temporary)')
+    parser.add_argument('--bert-id', type=str, default='bert-base-uncased')
+    parser.add_argument(
+        '--add-transformer-layer',
+        type="bool",
+        default=False,
+        help="Also add a transformer layer on top of Bert")
+    parser.add_argument('--pull-from-layer', type=int, default=-1,
```

Comment:
this never worked out right? Maybe we should just drop this?

Reply:
Like, it might be a little more memory friendly if this is dropped, but I'm not sure.